tests/pats.rs

```diff
@@ -0,0 +1,292 @@
+use actix_web::test;
+use chrono::{Duration, Utc};
+use common::database::*;
+use labrinth::models::pats::Scopes;
+use serde_json::json;
+
+use crate::common::environment::TestEnvironment;
+
+// importing common module.
+mod common;
+
+// Full pat test:
+// - create a PAT and ensure it can be used for the scope
+// - ensure access token is not returned for any PAT in GET
+// - ensure PAT can be patched to change scopes
+// - ensure PAT can be patched to change expiry
+// - ensure expired PATs cannot be used
+// - ensure PATs can be deleted
+#[actix_rt::test]
+pub async fn pat_full_test() {
```

Comment:
Just a general comment, not looking for an actual change - I hope that we can progress, with time, towards smaller tests that get closer to testing a small slice of functionality. Smaller tests makes it easier to diagnose test failures at a glance. If one of the early asserts in this test fails, then we get no information about the state of the other things it's testing.

However, given the current overhead of setting up for a test, and the sequential nature of these operations, this seems like a great place to start.

Reply:
Yeah reasonable.  Theres something of a trade off with the test size that is worth playing with in the future